quint/src/effects/base.ts

```diff
@@ -177,11 +177,36 @@ function bindEffect(name: string, effect: Effect): Either<string, Substitutions>
   }
 }
 
-function bindEntity(name: string, entity: Entity): Either<string, Substitutions> {
-  if (entityNames(entity).includes(name)) {
-    return left(`Can't bind ${name} to ${entityToString(entity)}: cyclical binding`)
-  } else {
-    return right([{ kind: 'entity', name, value: entity }])
+function bindEntity(name: string, entity: Entity): Substitutions {
+  switch (entity.kind) {
+    case 'concrete':
+    case 'variable':
+      return [{ kind: 'entity', name, value: entity }]
```

Comment:
Don't we need to check if `name === entity.name` here to avoid cyclical bindings?

Reply:
I don't think so. `a =.= a` is ok. It just unifies to an empty substitution. Cyclicality is only a problem when a variable is to unify with a term that includes itself as a (strict) sub-term.

But if you can think of a case that we should be testing for but aren't, I'll add a test to see if we can trigger some problem here.

Reply:
Ah, I see! So there is no such thing as cyclical bindings in our effect system anymore, right?

Reply:
Not at the level of "entity" variables, because they can only stand for sets of state variables. But there might be something that can happy with effect variables. I haven't looked at that logic as part of this. Could be that we don't have rich enough structures to make that possible tho?

Reply:
Oh, right. At the effect level we have operators, so we should have cyclical bindings in things like `e0 =.= (e0) => Read['x']`.

Thanks for helping me get this context back in and understanding the scope of your solution!

Reply:
Thanks for talking it thru with me and for the review! That is exactly right w/r/t effect variables. I think what we have here is just a special case due to simplicity of the "entity variables".